src/slot2/GBACart.cpp

```diff
@@ -0,0 +1,70 @@
+/*
+    Copyright (c) 2019 Adrian "asie" Siekierka
+
+    This file is part of melonDS.
+
+    melonDS is free software: you can redistribute it and/or modify it under
+    the terms of the GNU General Public License as published by the Free
+    Software Foundation, either version 3 of the License, or (at your option)
+    any later version.
+
+    melonDS is distributed in the hope that it will be useful, but WITHOUT ANY
+    WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+    FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with melonDS. If not, see http://www.gnu.org/licenses/.
+*/
+
+#include <stdio.h>
+#include <string.h>
+#include "../Platform.h"
+#include "GBACart.h"
+#include "GBACartNone.h"
+
+#define RAM_TO_CART(addr) (((addr) >> 1) & 0x00FFFFFF)
```

Comment:
consider using an inline function

Reply:
I'm not sure - I always preferred using defines for this kind of simple mathematical kludgery.